src/lib/storage/run_length_column/run_length_encoder.hpp

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <type_traits>
+
+#include "storage/base_column_encoder.hpp"
+
+#include "storage/run_length_column.hpp"
+#include "storage/value_column.hpp"
+#include "storage/value_column/value_column_iterable.hpp"
+#include "types.hpp"
+#include "utils/enum_constant.hpp"
+
+namespace opossum {
+
+class RunLengthEncoder : public ColumnEncoder<RunLengthEncoder> {
+ public:
+  static constexpr auto _encoding_type = enum_c<EncodingType, EncodingType::RunLength>;
+  static constexpr auto _uses_zero_suppression = false;
+
+  template <typename T>
+  std::shared_ptr<BaseEncodedColumn> _on_encode(const std::shared_ptr<const ValueColumn<T>>& value_column) {
```

Comment:
Somewhere above, you are asserting that the vector is not empty. This seems to accept an empty column.

Reply:
Yes addressed.